template/tsconfig/vue-i18n/i18n.d.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * you need to import the some interfaces
+ */
+import { DefineLocaleMessage } from 'vue-i18n'
+
+declare module 'vue-i18n' {
+  // define the locale messages schema
+  export interface DefineLocaleMessage {
+    hello: string
+  }
+}
```

Comment:
I don't think this file should be necessary? With TypeScript, you need to do something like:

```ts
import { createI18n } from 'vue-i18n'
import enUS from './locales/en-US.json'

// Type-define 'en-US' as the master schema for the resource
type MessageSchema = typeof enUS

const i18n = createI18n<[MessageSchema], 'en-US'>({
  locale: 'en-US',
  messages: {
    'en-US': enUS
  }
})
```